src/data/actions/assets.js

```diff
@@ -24,51 +16,120 @@ export const requestingAssets = () => ({
   type: assetActions.request.REQUESTING_ASSETS,
 });
 
-export const getAssets = (request, courseDetails) =>
+export const getAssets = (parameters, courseDetails) =>
   (dispatch, getState) => {
     dispatch(requestingAssets());
-    return clientApi.requestAssets(courseDetails.id, {
-      page: request.page,
-      assetTypes: request.assetTypes,
-      sort: request.sort,
-      direction: request.direction,
-    })
+
+    const state = getState();
+    const requestParameters = {
+      page: state.metadata.pagination.page,
+      assetTypes: state.metadata.filters.assetTypes,
+      sort: state.metadata.sort.sort,
+      direction: state.metadata.sort.direction,
+      ...parameters,
+    };
+
+    return clientApi.requestAssets(courseDetails.id, { ...requestParameters })
       .then((response) => {
         if (response.ok) {
           return response.json();
         }
         throw new Error(response);
       })
-      .then((json) => {
-        const lastRequest = getState().request;
-        if (isSameResponse(request, lastRequest)) {
-          dispatch(requestAssetsSuccess(json));
-        }
-      })
-      .catch((error) => {
-        dispatch(requestAssetsFailure(error));
-      });
+      .then(json => (
```

Comment:
I believe that we no longer have potential for an infinite loop with the removal of the request state, since there is no longer a piece of state to which changes cause API calls that, in turn, cause changes to that very same state.

As far as race conditions go, I was reading a little about dealing with race conditions in React/Redux, and it sounds like our state organization was just [suboptimal](https://stackoverflow.com/questions/37738268/how-to-avoid-race-conditions-when-fetching-data-with-redux), and it should not be a problem. I'm not sure whether the response in that post maps perfectly to our state, though.

TODO: I have to give it a bit more thought. I feel like it's possible for, between two API calls A1 and A2, for A2 to return before A1. In that case, what happens to the state? Can I replicate this situation somehow?